### ISSUE TITLE
Error in Movielens example?

### ISSUE THREAD

**Issue by bkj** [lyst/lightfm#312]
In the [movielens example](https://github.com/lyst/lightfm/tree/master/examples/movielens), should we be doing 
```
train_precision = precision_at_k(model, train, k=10).mean()
test_precision = precision_at_k(model, test, k=10, train_interactions=train).mean()

train_auc = auc_score(model, train).mean()
test_auc = auc_score(model, test, train_interactions=train).mean()
```
I haven't looked through all the code, but I think if `train_interactions=None` then most of the top K recommendations are going to be movies that the user has viewed in the training set, and thus not in the test set.

Making this change bumps test precision from `0.11` to `0.11` and test AUC from `0.90` to `0.93`.

~ Ben

**Comment by bkj** [comment]
Also -- Are there any benchmarks for "good" P@K on Movielens?  I'm trying to figure out what the state-of-the-art on these benchmarks is, but can't find many resources.

~ Ben

**Comment by maciejkula** [comment]
Yes, that makes sense. I would accept the fix if you'd like to make a PR!

There aren't really any benchmarks in the same way you'd find benchmarks for imagenet, for example. I think it would be a worthwhile project to set up a repo that does benchmarking of the sort.

**Comment by maciejkula** [comment]
Merged the fix.
